File: Services/Dto/CompanyDto.cs
using Services.Dto.Shared;

namespace Services.Dto
{
    public class CompanyDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int AddressId { get; set; }
        public AddressDto Address { get; set; }
        public int ContactId { get; set; }
        public ContactDto Contact { get; set; }
    }


}

File: Views/src/shared/dtos/company-dto.ts
import { StockDto } from "src/components/product/dtos/stock-dto";
import { AddressDto } from "./address-dto";
import { ContactDto } from "./contact-dto";

export class CompanyDto {
   id:number ;
   name:string ;
   stock:StockDto;
   address:AddressDto ;
   contact:ContactDto ;
}
